sdk/java/pulumi/src/main/java/com/pulumi/provider/internal/infer/ComponentAnalyzer.java

```diff
@@ -50,10 +50,19 @@ public static PackageSpec generateSchema(Class<?>... classes) {
         String name = classes[0].getPackage().getName();
         // Use last segment of package name if it contains dots
         name = name.substring(name.lastIndexOf('.') + 1);
-        
+
         return generateSchema(new Metadata(name, null, null), classes);
     }
 
+    private static String getNamespace(Class<?> clazz) {
+        String[] split = clazz.getPackage().getName().split("\\.");
+        if (split.length == 0) {
+            return "";
+        }
+        return split[1];
```

Comment:
This will fail for length == 1. Unit tests, please.

Reply:
Done.